zbus/src/address/transport.rs

```diff
@@ -0,0 +1,104 @@
+//! D-Bus supported transports.
+
+use std::fmt;
+
+use super::{
+    address::{KeyValFmt, KeyValFmtAdd},
+    percent, Address, Error, Result,
+};
+
+mod autolaunch;
+pub use autolaunch::{Autolaunch, AutolaunchScope};
+
+mod launchd;
+pub use launchd::Launchd;
+
+mod nonce_tcp;
+pub use nonce_tcp::NonceTcp;
+
+mod systemd;
+pub use systemd::Systemd;
+
+mod tcp;
+pub use tcp::{Tcp, TcpFamily};
+
+mod unix;
+pub use unix::{Unix, UnixAddrKind};
+
+mod unixexec;
+pub use unixexec::Unixexec;
+
+#[cfg(feature = "vsock_address")]
+mod vsock;
+#[cfg(feature = "vsock_address")]
+pub use vsock::Vsock;
+
+/// A D-Bus transport.
+#[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum Transport<'a> {
+    Unix(unix::Unix<'a>),
+    Launchd(launchd::Launchd<'a>),
+    Systemd(systemd::Systemd<'a>),
+    Tcp(tcp::Tcp<'a>),
+    NonceTcp(nonce_tcp::NonceTcp<'a>),
```

Comment:
In 4.0 I merged this with Tcp and I think that's the right approach.

This reminds me, have you made an effort to ensure all the changes from 3.x to 4.x (that you didn't object to), have been incorporated?

Reply:
They are treated as different transport in the spec. For consistency, it's best tofollow the spec, otherwise it's surprising and people may wonder if nonce-tcp is actually supported. 

Which changes are you talking about? Yes, I looked at your implementation and we have the same tests.. (except duplicate keys you ask me to drop)

Reply:
I didn't revert.. I wrote this first, based on 3.x. You changed it after, I didn't follow, as I think it could lead to further issues.